src/main/java/com/unzer/payment/service/PaymentService.java

```diff
@@ -256,28 +259,30 @@ private JsonIdObject getJsonObjectFromTypeId(String typeId) {
       case PF_EFINANCE:
       case UNZER_PAYLATER_INVOICE:
       case KLARNA:
-        return new JsonIdObject();
+        return new ApiIdObject();
       case PAYPAL:
-        return new JsonPaypal();
+        return new ApiPaypal();
       case CARD:
-        return new JsonCard();
+        return new ApiCard();
       case IDEAL:
-        return new JsonIdeal();
+        return new ApiIdeal();
       case SEPA_DIRECT_DEBIT:
       case SEPA_DIRECT_DEBIT_GUARANTEED:
       case SEPA_DIRECT_DEBIT_SECURED:
-        return new JsonSepaDirectDebit();
+        return new ApiSepaDirectDebit();
```

Comment:
If all direct debit types share the same api model then "PAYLATER_DIRECT_DEBIT" is the 
'same' and could be added here.

Reply:
Yes, and what I mean is that the instatiate the same class 'ApiSepaDirectDebit()'  and therefore it could be moved together like:

      case SEPA_DIRECT_DEBIT:  

      case SEPA_DIRECT_DEBIT_GUARANTEED:

      case SEPA_DIRECT_DEBIT_SECURED:

      case PAYLATER_DIRECT_DEBIT:

        return new ApiSepaDirectDebit();

However, just a suggestion.